### PR TITLE
Fixes #36828 - add host products report jail methods

### PR DESCRIPTION
**Pull request by ianballou**
#### What are the changes introduced in this pull request?
Adds some jail methods for https://github.com/theforeman/foreman/pull/9863

#### Considerations taken when implementing this change?
These should be perfectly safe methods.

#### What are the testing steps for this pull request?
Test out https://github.com/theforeman/foreman/pull/9863 with this PR.

### CODE REVIEW

**Comment by ianballou** [comment]
Also adds a new method to list host installed products with IDs

File: app/models/katello/concerns/host_managed_extensions.rb [hunk 1]
```diff
@@ -587,7 +587,7 @@ class ::Host::Managed::Jail < Safemode::Jail
         :host_collections, :pools, :hypervisor_host, :installed_debs,
         :installed_packages, :traces_helpers, :advisory_ids, :package_names_for_job_template,
         :filtered_entitlement_quantity_consumed, :bound_repositories,
-        :single_content_view, :single_lifecycle_environment
+        :single_content_view, :single_lifecycle_environment, :purpose_role_status_label, :purpose_usage_status_label
```
Comment (jeremylenz) [inline app/models/katello/concerns/host_managed_extensions.rb:590]: see https://github.com/theforeman/foreman/pull/9863#discussion_r1364284473

**Review by jeremylenz — APPROVED** [review]

**Comment by jeremylenz** [comment]
[test katello]

**Comment by ianballou** [comment]
[test katello]

**Comment by jeremylenz** [comment]
I think we're gonna have to wait for https://github.com/Katello/katello/pull/10774 to be merged ;)

**Pull request merged**